tests/test_detectAndMeasure.py

```diff
@@ -621,16 +626,17 @@ def _detection_wrapper(positive=True):
             else:
                 difference.maskedImage -= transients.maskedImage
             score = subtractTask._convolveExposure(difference, scienceKernel, subtractTask.convolutionControl)
-            output = detectionTask.run(science, matchedTemplate, difference, score)
+            # NOTE: NoiseReplacer (run by forcedMeasurement) can modify the
+            # science image if we've e.g. removed parents post-deblending.
+            # Pass a clone of the science image, so that it doesn't disrupt
+            # later tests.
+            output = detectionTask.run(science.clone(), matchedTemplate, difference, score)
             refIds = []
             scale = 1. if positive else -1.
-            goodSrcFlags = subtractTask._checkMask(score.mask, transientSources,
-                                                   subtractTask.config.badMaskPlanes)
+            # sources near the edge may have untrustworthy centroids
+            goodSrcFlags = ~output.diaSources['base_PixelFlags_flag_edge']
```

Comment:
This is interesting! We should create a ticket to improve our centroids near the edge. They should be able to ignore those pixels, but I see this test fails without this check with one source being off by 5 pixels.